src/clnt_rdma.c

```diff
@@ -274,14 +266,16 @@ clnt_rdma_call(CLIENT *cl,		/* client handle */
 static void
 clnt_rdma_geterr(CLIENT *cl, struct rpc_err *errp)
 {
-	struct cm_data *cm = CM_DATA((struct cx_data *) cl->cl_p1);
+	struct cm_data *cm = CM_DATA(CX_DATA(cl));
```

Comment:
Because in some places there is only a reference to the per-union variables, and in others there are also references to the CX variables.  Also, in various cases one of the lines will be replaced in a future patch.

And are we really short on stack variables that the compiler will optimize away anyway? Hint: CLIENT is the first element, so it's just a cast.